submit_cgap/tests/test_show_upload_info.py

```diff
@@ -0,0 +1,52 @@
+from unittest import mock
+from ..scripts.show_upload_info import main as show_upload_info_main
+from ..scripts import show_upload_info as show_upload_info_module
+
+
+def test_show_upload_info_script():
+
+    def test_it(args_in, expect_exit_code, expect_called, expect_call_args=None):
+        output = []
+        with mock.patch.object(show_upload_info_module, "print") as mock_print:
+            mock_print.side_effect = lambda *args: output.append(" ".join(args))
+            with mock.patch.object(show_upload_info_module, "show_upload_info") as mock_show_upload_info:
+                try:
+                    show_upload_info_main(args_in)
+                    mock_show_upload_info.assert_called_with(**expect_call_args)
+                except SystemExit as e:
+                    assert e.code == expect_exit_code
+                assert mock_show_upload_info.call_count == (1 if expect_called else 0)
+                assert output == []
+
+    test_it(args_in=[], expect_exit_code=2, expect_called=False)  # Missing args
+    test_it(args_in=['some-guid'], expect_exit_code=0, expect_called=True, expect_call_args={
```

Comment:
Advantage of this structure vs. parametrizing? I wouldn't be against adding a dependency like the following if we want to stick with unittest as I really like this feature of Pytest: https://github.com/wolever/parameterized.

Reply:
It lets you set up the environment in which the parameterization is to occur, although in this case it doesn't matter, I think. Sometimes it does. I've found the parameterization feature to be sometimes helpful, sometimes not, but it's not wired into my brain so sometimes I forget to do it. But this way you do get things like editor support for completion of argument names so you can keyword-call them, etc.